qa/acceptance-tests/src/test/java/io/camunda/it/client/JobSearchTest.java

```diff
@@ -0,0 +1,659 @@
+
+    // then
+    assertThat(resultAsc.items())
+        .extracting(Job::getElementInstanceKey)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.naturalOrder()).toList());
+    assertThat(resultDesc.items())
+        .extracting(Job::getElementInstanceKey)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.reverseOrder()).toList());
+  }
+
+  @Test
+  void shouldSortJobsByTenantId() {
+    // given
+    final var all =
+        camundaClient.newJobSearchRequest().send().join().items().stream()
+            .map(Job::getTenantId)
+            .toList();
+    // when
+    final var resultAsc =
+        camundaClient.newJobSearchRequest().sort(s -> s.tenantId().asc()).send().join();
+    final var resultDesc =
+        camundaClient.newJobSearchRequest().sort(s -> s.tenantId().desc()).send().join();
+
+    // then
+    assertThat(resultAsc.items())
+        .extracting(Job::getTenantId)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.naturalOrder()).toList());
+    assertThat(resultDesc.items())
+        .extracting(Job::getTenantId)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.reverseOrder()).toList());
+  }
+
+  @Test
+  void shouldSearchByFromLimit() {
+    // given
+    final var allJobs =
+        camundaClient.newJobSearchRequest().send().join().items().stream()
+            .map(Job::getJobKey)
+            .toList();
+    final int limit = 2;
+
+    // when
+    final var result = camundaClient.newJobSearchRequest().page(p -> p.limit(limit)).send().join();
+
+    // then
+    assertThat(result.items()).hasSize(2);
+    assertThat(result.items()).extracting(Job::getJobKey).containsAll(allJobs.subList(0, 2));
+  }
+
+  @Test
+  void shouldSearchAfterSecondItem() {
+    // given
+    final var allJobs = camundaClient.newJobSearchRequest().send().join().items();
+    final var secondItemKey = allJobs.get(1).getJobKey();
+
+    // when
+    final var result =
+        camundaClient
+            .newJobSearchRequest()
+            .page(p -> p.searchAfter(Collections.singletonList(secondItemKey)))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items())
+        .extracting(Job::getJobKey)
+        .containsExactlyElementsOf(
+            allJobs.stream().map(Job::getJobKey).toList().subList(2, allJobs.size()));
+  }
+
+  @Test
+  void shouldSearchBeforeSecondItem() {
+    // given
+    final var allJobs = camundaClient.newJobSearchRequest().send().join().items();
+    final var secondItemKey = allJobs.get(1).getJobKey();
+
+    // when
+    final var result =
+        camundaClient
+            .newJobSearchRequest()
+            .page(p -> p.searchBefore(Collections.singletonList(secondItemKey)))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items())
+        .extracting(Job::getJobKey)
+        .containsExactlyElementsOf(allJobs.stream().map(Job::getJobKey).toList().subList(0, 1));
+  }
```

Comment:
💭 This will change a bit with https://github.com/camunda/camunda/pull/33026. We can either leave it out here completely or adjust it when that other PR was merged (or adjust that PR when this is merged first).